### ISSUE TITLE
check doesn't respect runxfail option

### ISSUE THREAD

**Issue by taylermulligan** [okken/pytest-check#171]
When `runxfail` option is set (`pytest --runxfail ...`) tests that use `check` are reported as xfail instead of failed:

_xfail_test.py_
```python
import pytest

@pytest.mark.xfail(reason="foo")
def test_xfail(check):
    with check:  # with this line gone, this is actual behaviour
        assert False
``` 

# Expected behaviour

```bash
$ pytest --runxfail xfail_test.py 
================================================================ test session starts =================================================================
platform linux -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/taylermulligan/pytest_xfail
plugins: check-2.4.1
collected 1 item                                                                                                                                     

xfail_test.py F                                                                                                                                [100%]

====================================================================== FAILURES ======================================================================
_____________________________________________________________________ test_xfail _____________________________________________________________________

check = <pytest_check.context_manager.CheckContextManager object at 0x7f2db410d120>

    @pytest.mark.xfail(reason="foo")
    def test_xfail(check):
>       assert False
E       assert False

xfail_test.py:5: AssertionError
============================================================== short test summary info ===============================================================
FAILED xfail_test.py::test_xfail - assert False
================================================================= 1 failed in 0.01s ==================================================================
```

# Actual behaviour

```bash
$ pytest --runxfail xfail_test.py 
================================================================ test session starts =================================================================
platform linux -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/taylermulligan/pytest_xfail
plugins: check-2.4.1
collected 1 item                                                                                                                                     

xfail_test.py x                                                                                                                                [100%]

================================================================= 1 xfailed in 0.03s =================================================================
```
